addons/crm/models/crm_lead.py

```diff
@@ -1466,20 +1479,60 @@ def _cron_update_automated_probabilities(self):
             pending_lead_domain = ['&', '&', ('stage_id', '!=', False), ('create_date', '>', pls_start_date),
                                    '|', ('probability', '=', False), '&', ('probability', '<', 100), ('probability', '>', 0)]
             leads_to_update = self.env['crm.lead'].search(pending_lead_domain)
```

Comment:
Not sure, but: shouldn't we turn off prefetching, prepare a list of record sets based on batch_step, and prefetch only by record sets ? Or due to the orm being smart it already works that way ?